src/libaktualizr-posix/ipuptanesecondary.h

```diff
@@ -7,22 +7,39 @@
 #include "uptane/secondaryinterface.h"
 
 namespace Uptane {
+
 class IpUptaneSecondary : public SecondaryInterface {
  public:
-  explicit IpUptaneSecondary(const SecondaryConfig& conf) : SecondaryInterface(conf) {}
+  static std::shared_ptr<Uptane::SecondaryInterface> create(const std::string& address, unsigned short port);
+
+  explicit IpUptaneSecondary(const sockaddr_in& sock_addr, EcuSerial serial, HardwareIdentifier hw_id,
+                             PublicKey pub_key);
+
+  // what this method for ? Looks like should be removed out of SecondaryInterface
+  void Initialize() override{};
+
+  // It looks more natural to return const EcuSerial& and const Uptane::HardwareIdentifier&
+  // and they should be 'const' methods
```

Comment:
You might be right. In the past we didn't do that in case system calls or other non-const-friendly operations were required to implement these. We can re-evaluate that again soon.